mbedtls/src/error.rs

```diff
@@ -31,9 +31,15 @@ pub const ERR_UTF8_INVALID: c_int = -0x10000;
 
 macro_rules! error_enum {
     {enum $n:ident {$($rust:ident = $c:ident,)*}} => {
+
         #[derive(Debug, Eq, PartialEq)]
         pub enum $n {
             $($rust,)*
+        }
+
+        #[derive(Debug, Eq, PartialEq)]
+        pub enum Error {
+            MbedError(MbedErrorCode, c_int),
```

Comment:
This doesn't make sense to me, how can I pull the 2 different errors out of this?

Reply:
changed to 3 types of Error enum variants: HighLevel, LowLevel, HighAndLowLevel